shell/platform/android/test/io/flutter/plugin/platform/PlatformPluginTest.java

```diff
@@ -321,49 +323,80 @@ public void setSystemUiMode() {
     when(fakeActivity.getWindow()).thenReturn(fakeWindow);
     PlatformChannel fakePlatformChannel = mock(PlatformChannel.class);
     PlatformPlugin platformPlugin = new PlatformPlugin(fakeActivity, fakePlatformChannel);
+    WindowInsetsController fakeWindowInsetsController = mock(WindowInsetsController.class);
+    when(fakeWindow.getInsetsController()).thenReturn(fakeWindowInsetsController);
 
     if (Build.VERSION.SDK_INT >= 28) {
       platformPlugin.mPlatformMessageHandler.showSystemUiMode(
           PlatformChannel.SystemUiMode.LEAN_BACK);
-      verify(fakeDecorView)
```

Comment:
In the implementations of `WindowInsetsControllerCompat` and `WindowCompat`, each of the system UI modes are accomplished through a couple of different `setSystemUiVisibility(...)` calls for SDK < 30, so I had to modify the test slightly for those versions.